src/main/java/me/clip/placeholderapi/expansion/manager/LocalExpansionManager.java

```diff
@@ -393,7 +410,15 @@ private void registerAll(@NotNull final CommandSender sender) {
 
   private void unregisterAll() {
     for (final PlaceholderExpansion expansion : Sets.newHashSet(expansions.values())) {
-      if (expansion.persist()) {
+      if (expansion.persist() || expansion.getExpansionType() == Type.INTERNAL) {
```

Comment:
Can't we not skip external expansions here, and just skip them in the uregister(PlaceholderExpansion) method? This way we don't have duplicate code. This would mean, we would only skip here if expansion type is internal.

Reply:
I decided to skip unregister for external if persist is true, because maybe the author didn't understand stuff and decided to override the expansion type when they shouldn't? Causing this particular situation of an external expansion being set to be persistent...

Perhaps we could make it different by still unregister external expansions, no matter the persist setting, but print a warn about this not allowed combination to inform about a possible misconfiguration.